### ISSUE TITLE
Post-commit testing fails on compilation of vc-intrinsics

### ISSUE THREAD

**Issue by alexbatashev** [intel/llvm#3142]
Compilation fails during link phase:

```
FAILED: lib/libLLVMGenXIntrinsics.so.12git 
: && /usr/bin/clang++-13 -fPIC -fPIC -fvisibility-inlines-hidden -Werror=date-time -Werror=unguarded-availability-new -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wmissing-field-initializers -pedantic -Wno-long-long -Wimplicit-fallthrough -Wcovered-switch-default -Wno-noexcept-type -Wnon-virtual-dtor -Wdelete-non-virtual-dtor -Wsuggest-override -Wstring-conversion -fdiagnostics-color -ffunction-sections -fdata-sections -O3  -Wl,-z,defs -Wl,-z,nodelete   -Wl,-rpath-link,/home/runner/work/llvm/llvm/build/./lib  -Wl,-O3 -Wl,--gc-sections -shared -Wl,-soname,libLLVMGenXIntrinsics.so.12git -o lib/libLLVMGenXIntrinsics.so.12git _deps/vc-intrinsics-build/GenXIntrinsics/lib/GenXIntrinsics/CMakeFiles/LLVMGenXIntrinsics.dir/GenXIntrinsics.cpp.o _deps/vc-intrinsics-build/GenXIntrinsics/lib/GenXIntrinsics/CMakeFiles/LLVMGenXIntrinsics.dir/GenXRestoreIntrAttr.cpp.o _deps/vc-intrinsics-build/GenXIntrinsics/lib/GenXIntrinsics/CMakeFiles/LLVMGenXIntrinsics.dir/GenXSimdCFLowering.cpp.o _deps/vc-intrinsics-build/GenXIntrinsics/lib/GenXIntrinsics/CMakeFiles/LLVMGenXIntrinsics.dir/GenXSingleElementVectorUtil.cpp.o _deps/vc-intrinsics-build/GenXIntrinsics/lib/GenXIntrinsics/CMakeFiles/LLVMGenXIntrinsics.dir/GenXSPIRVReaderAdaptor.cpp.o _deps/vc-intrinsics-build/GenXIntrinsics/lib/GenXIntrinsics/CMakeFiles/LLVMGenXIntrinsics.dir/GenXSPIRVWriterAdaptor.cpp.o  -Wl,-rpath,"\$ORIGIN/../lib"  lib/libLLVMCodeGen.so.12git  lib/libLLVMCore.so.12git  lib/libLLVMSupport.so.12git  -Wl,-rpath-link,/home/runner/work/llvm/llvm/build/lib && :
_deps/vc-intrinsics-build/GenXIntrinsics/lib/GenXIntrinsics/CMakeFiles/LLVMGenXIntrinsics.dir/GenXSingleElementVectorUtil.cpp.o: In function `llvm::genx::collapseExtractInstructions(llvm::Function&, bool)':
GenXSingleElementVectorUtil.cpp:(.text._ZN4llvm4genxL27collapseExtractInstructionsERNS_8FunctionEb+0x10b): undefined reference to `llvm::SimplifyExtractElementInst(llvm::Value*, llvm::Value*, llvm::SimplifyQuery const&)'
GenXSingleElementVectorUtil.cpp:(.text._ZN4llvm4genxL27collapseExtractInstructionsERNS_8FunctionEb+0x1b4): undefined reference to `llvm::SimplifyExtractElementInst(llvm::Value*, llvm::Value*, llvm::SimplifyQuery const&)'
_deps/vc-intrinsics-build/GenXIntrinsics/lib/GenXIntrinsics/CMakeFiles/LLVMGenXIntrinsics.dir/GenXSingleElementVectorUtil.cpp.o: In function `llvm::genx::collapseInsertInstructions(llvm::Function&, bool)':
GenXSingleElementVectorUtil.cpp:(.text._ZN4llvm4genxL26collapseInsertInstructionsERNS_8FunctionEb+0xff): undefined reference to `llvm::SimplifyInsertElementInst(llvm::Value*, llvm::Value*, llvm::Value*, llvm::SimplifyQuery const&)'
_deps/vc-intrinsics-build/GenXIntrinsics/lib/GenXIntrinsics/CMakeFiles/LLVMGenXIntrinsics.dir/GenXSingleElementVectorUtil.cpp.o: In function `llvm::genx::collapseBitcastInstructions(llvm::Function&, bool)':
GenXSingleElementVectorUtil.cpp:(.text._ZN4llvm4genxL27collapseBitcastInstructionsERNS_8FunctionEb+0xe2): undefined reference to `llvm::SimplifyCastInst(unsigned int, llvm::Value*, llvm::Type*, llvm::SimplifyQuery const&)'
GenXSingleElementVectorUtil.cpp:(.text._ZN4llvm4genxL27collapseBitcastInstructionsERNS_8FunctionEb+0x1ef): undefined reference to `llvm::SimplifyCastInst(unsigned int, llvm::Value*, llvm::Type*, llvm::SimplifyQuery const&)'
```

Full logs can be found here: https://github.com/intel/llvm/runs/1813851231?check_suite_focus=true

**Comment by DenisBakhvalov** [comment]
I submitted a PR with the fix here: https://github.com/intel/vc-intrinsics/pull/49

**Comment by kbobrovs** [comment]
> I submitted a PR with the fix here: [intel/vc-intrinsics#49](https://github.com/intel/vc-intrinsics/pull/49)

@DenisBakhvalov, I suggest to add reviewers in the above PR and/or notify code owners

**Comment by DenisBakhvalov** [comment]
@kbobrovs I can't find a way to do this. I just wrote a comment and tagged a few maintainers.
